src/Extensions/Logging/Internal/OmexLogsEventSource.cs

```diff
@@ -0,0 +1,198 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.Diagnostics.Tracing;
+using System.Globalization;
+using System.Text;
+using Microsoft.Extensions.Logging;
+using Microsoft.Omex.Extensions.Abstractions;
+
+namespace Microsoft.Omex.Extensions.Logging
+{
+	[EventSource(Name = "Microsoft-OMEX-Logs")]
+	internal sealed class OmexLogsEventSource : EventSource
+	{
+		static OmexLogsEventSource()
+		{
+			Process process = Process.GetCurrentProcess();
+			ProcessName = string.Format("{0} (0x{1:X4})", process.ProcessName, process.Id);
+		}
+
+
+		private static string ProcessName { get; }
+
+
+		public OmexLogsEventSource(IMachineInformation machineInformation, IServiceContext context)
+		{
+			m_machineInformation = machineInformation;
+			m_serviceContext = context;
+		}
+
+
+		private readonly IMachineInformation m_machineInformation;
+		private readonly IServiceContext m_serviceContext;
+
+		private string ApplicationName => m_machineInformation.MachineRole;
+		private string ServiceName => m_machineInformation.ServiceName;
+		private string AgentName => m_machineInformation.AgentName;
+		private string BuildVersion => m_machineInformation.BuildVersion;
+
+		private Guid PartitionId => m_serviceContext?.PartitionId ?? Guid.Empty;
+		private long ReplicaId => m_serviceContext?.ReplicaOrInstanceId ?? 0;
+
+		private string CorrelationId => Activity.Current?.Id ?? string.Empty; //Breaking Change: Type Changed from Guid ?? Guid.Empty
+		private string TransactionId => (Activity.Current?.TraceId ?? default).ToHexString(); //Breaking Change: Type Changed from uint ?? 0u
+
+
+		[NonEvent]
+		public void ServiceMessage(string message, LogLevel level, string category, EventId eventId, int threadId)
+		{
+			if (!IsEnabled())
+			{
+				return;
+			}
+
+			// Both converts should be done more efficient
```

Comment:
>// Both converts should be done more efficient [](start = 3, length = 46)

please create an Issue with a brief description how the conversion can be optimized

Reply:
Makes sense, thank you, I'll create issue before merging

Reply:
I'll leave it like this for now since I'm unable to come out with better way of doing convert, I hope for future version we'll just replace tagging method